sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java

```diff
@@ -296,13 +292,51 @@ private SourceAndVersion prepareUpdate(DocTableInfo tableInfo,
                 indexShard.shardId(), Constants.DEFAULT_MAPPING_TYPE, item.id(), "TODO: add explanation");
         }
 
-        Tuple<XContentType, Map<String, Object>> sourceAndContent = XContentHelper.convertToMap(getResult.internalSourceRef(), true);
-        final Map<String, Object> updatedSourceAsMap;
-        final XContentType updateSourceContentType = sourceAndContent.v1();
+        return getResult;
+    }
+
+    private List<Input<?>> resolveSymbols(ReferenceResolver<CollectExpression<GetResult, ?>> referenceResolver,
+                                          GetResult getResult,
+                                          List<? extends Symbol> updateAssignments,
```

Comment:
the `updateAssignments` could be changed to a `Iterable` - so that in `processGeneratedColumns` building a list becomes unnecessary.